app/models/ocelot/stanzas/ChoiceStanza.scala

```diff
@@ -53,15 +49,65 @@ object ChoiceStanza {
   implicit val reads: Reads[ChoiceStanza] =
     (
       (JsPath \ "next").read[Seq[String]](minLength[Seq[String]](2)) and
-      (JsPath \ "tests").read[Seq[ChoiceTest]](minLength[Seq[ChoiceTest]](1)) and
+      (JsPath \ "tests").read[Seq[ChoiceStanzaTest]](minLength[Seq[ChoiceStanzaTest]](1)) and
       (JsPath \ "stack").read[Boolean]
     )(ChoiceStanza.apply _)
 
   implicit val writes: OWrites[ChoiceStanza] =
     (
       (JsPath \ "next").write[Seq[String]] and
-      (JsPath \ "tests").write[Seq[ChoiceTest]] and
+      (JsPath \ "tests").write[Seq[ChoiceStanzaTest]] and
       (JsPath \ "stack").write[Boolean]
     )(unlift(ChoiceStanza.unapply))
 
+}
+
+sealed trait ChoiceTest {
+  val left: String
+  val right: String
+  def eval(labels: Labels): Boolean
+  def value(arg: String, labels: Labels): String = labelReference(arg).fold(arg)(ref => labels.value(ref).getOrElse(""))
+  def op(f: (BigDecimal, BigDecimal) => Boolean, g: (String, String) => Boolean, labels: Labels): Boolean =
```

Comment:
We may need a date function here eventually. Not certain about number; that may be handled by BigDecimal.